sympy/integrals/tests/test_transforms.py

```diff
@@ -583,6 +583,21 @@ def simp_hyp(expr):
         Matrix([[exp(t)*Heaviside(t), 0], [0, exp(2*t)*Heaviside(t)]])
 
 
+def test_inverse_laplace_transform_delta():
+    from sympy import DiracDelta
+    ILT = inverse_laplace_transform
+    t = symbols('t')
+    assert ILT(2, s, t) == 2*DiracDelta(t)
+    assert ILT(2*exp(3*s) - 5*exp(-7*s), s, t) == \
+        2*DiracDelta(t + 3) - 5*DiracDelta(t - 7)
+    a = cos(sin(7)/2)
+    assert ILT(a*exp(-3*s), s, t) == a*DiracDelta(t - 3)
+    assert ILT(exp(2*s), s, t) == DiracDelta(t + 2)
+    r = Symbol('r', real=True)
+    assert ILT(exp(r*s), s, t) == DiracDelta(t + r)
+    # TODO: what if r.is_real is None/False?  See meijerint_inversion
```

Comment:
Note the transform code which calls `meijerint_inversion` may not make great choices but I think what we return here is consistent with what the G function part of `meijerint_inversion`.

Reply:
`inverse_laplace_transform` does the right thing when `noconds=False` is passed.  I've added tests for this.

IMHO, it might be better if `inverse_laplace_transform` just gave Piecewise.  But that's a separate issue.

Reply:
> IMHO, it might be better if inverse_laplace_transform just gave Piecewise. But that's a separate issue.

+1